### PR TITLE
Update Appointment component - FHIR next

### PR DESCRIPTION
**Pull request by sflejszman**
Issue:  [ZH-15](https://app.zenhub.com/workspaces/patient-app-6140de7e31081800158d8ecf/issues/1uphealth/patient/15)

---

##### PR Checklist
- [x] Give this PR a meaningful title
- [x] Add a link to the related issue at the top of this description (above)
- [x] Connect this PR with the related issue via ZenHub with the button below this text box (or at the bottom of the page after the PR is created)
- [x] Ensure your branch is up to date with the target branch and resolve any conflicts
- [x] Answer the below questions to describe your PR for reviewers
- [x] Request at least two reviewers using the "Reviewers" section on the right, usually including at least one reviewer from your team
- [x] Notify the requested reviewers in the #code-review Slack channel once the PR is ready for review

---

## Why are these changes needed?
Update styles for Appointment according to designs in FIGMA.

## What changed?
- update styles for appointment component,
- update datatypes,
- fix Header component,
- update icon for Condition,
- Immunization and Procedure resource
- check lint
- update specs

## Screenshot
Screenshot 2021-11-19 at 13 36 34

## How are these changes tested?
- All automatic tests are passing
- The ResourceCategory component might be tested manually by viewing the current version of deployed [storybook](https://fhir-react-lib-test-storybook.s3.amazonaws.com/branch/update-fhir-component-appointment/index.html?path=/story/resourcecategory--example-without-items-count)

### CODE REVIEW

File: src/components/ui/index.js [hunk 1]
```diff
@@ -153,7 +154,7 @@ export const TableHeader = props => {
   return (
     <th
       className={`${expand && 'w-100'} ${noWordWrap &&
-        'text-nowrap'} text-muted`}
+        'text-nowrap'} text-muted text-gray-500`}
```
Comment (michal-maksajda) [inline src/components/ui/index.js:157]: My suggestion is to keep only one text style property. The other one here has no effect.

**Review by michal-maksajda — APPROVED** [review]
👍

File: src/style.scss [hunk 2]
```diff
@@ -61,7 +61,11 @@ $enable-negative-margins: true;
 }
 
 .value__label {
-  width: 20%;
+  width: 130px;
```
Comment (liam1uphealth) [inline src/style.scss:64]: this may be the cause for this component not being responsive. 
image

**Review by liam1uphealth — CHANGES_REQUESTED** [review]
going to block this until the responsive issues are addressed.

File: src/style.scss (hunk 2 above)
Comment (sflejszman) [reply src/style.scss:64]: Done 👍 
Screenshot 2021-11-23 at 14 57 23

**Review by jaceksanko — APPROVED** [review]
👍

**Review by glenn1uphealth — APPROVED** [review]

**Pull request merged**